# U.S. Medical Insurance Costs

### Region is a new field introduced in the project which hasn't appeared during the lessons. So, one of the things I would want to analyze is what are the nuances and trends when it comes to different regions. Where are people paying higher insurance costs and what could be the reason for it. Based on this, I would be able to summarize other insights after analyzing this data. Let's start by importing the csv file into the notebook.

In [7]:
#import the CSV Library
import csv

After looking at the insurances.csv data, we can confirm that we have 7 columns representing different attributes. Let us open the csv file and create empyt lists to load the respective fields with the data.

In [76]:
# Open the 'insurance.csv' file for reading
with open('insurance.csv', newline='') as insurance_csv:
    # Create a CSV DictReader to read the contents of the file as dictionaries
    records = csv.DictReader(insurance_csv)
    
    # Initialize empty lists to store data from the CSV
    age = []        # List to store age values
    sex = []        # List to store sex values
    bmi = []        # List to store BMI values
    children = []   # List to store children values
    smoker = []     # List to store smoker status values
    region = []     # List to store region values
    charges = []    # List to store insurance charges values
    
    # Iterate through each row in the CSV
    for row in records:
        # Append data from the current row to respective lists
        age.append(float(row['age']))             # Convert 'age' to float
        sex.append(row['sex'])                    # Store 'sex' as-is (string)
        bmi.append(float(row['bmi']))             # Convert 'bmi' to float
        children.append(float(row['children']))   # Convert 'children' to float
        smoker.append(row['smoker'])              # Store 'smoker' as-is (string)
        region.append(row['region'])              # Store 'region' as-is (string)
        charges.append(float(row['charges']))     # Convert 'charges' to float

# All of our data from the 'insurance.csv' file is now in their own lists.


Each of the above lists will contain the data from each of the respective columns

Now, let's try to analyze some basic information about the data.

1) Average Insurance Cost Overall

In [21]:
#defining a function to return the average cost
average_cost_overall = 0.0
def average_insurance_cost(charges):
    average_cost_overall = round(sum(charges)/len(charges),2)
    return average_cost_overall
print("The average insurance cost of the population is: " + str(average_insurance_cost(charges)))    

The average insurance cost of the population is: 13270.42


2. Average age Overall

In [23]:
#Calculating the average age accross the data
average_age = 0 
def average_age_calc(age):
    average_age = round(sum(age)/len(age)) 
    return average_age
print("The average age overall is " + str(average_age_calc(age)))
    

The average age overall is 39


3) Percentage of Male and Female population

In [27]:
#sex distribtion across the data 
num_males = 0
num_females=0
for i in sex:
    if i == "male":
        num_males+=1
    else:
        num_females+=1
print("The number of people in the study = " + str(len(sex)))
print("The number of males = " + str(num_males))
print("The number of females = " + str(num_females))
print("Percentage of males = " + str(round((num_males/len(sex))*100,2)) + " %")
print("Percentage of females = " + str(round((num_females/len(sex))*100,2)) + " %")

The number of people in the study = 1338
The number of males = 676
The number of females = 662
Percentage of males = 50.52 %
Percentage of females = 49.48 %


We can summarize that the data seems to be fair considering almost similar percentages of males vs females for the study.

Similarly, lets try to check the percentages of smoker vs non-smoker

In [77]:
#Smoker stats across the data
num_smoker = 0
num_nonsmoker = 0
for j in smoker:
    if j == "yes":
        num_smoker+=1
    else:
        num_nonsmoker+=1
print("The number of people in the study = " + str(len(smoker)))
print("The number of smokers = " + str(num_smoker))
print("The number of non-smokers = " + str(num_nonsmoker))
print("Percentage of smokers = " + str(round((num_smoker/len(smoker))*100,2)) + " %")
print("Percentage of non-smokers = " + str(round((num_nonsmoker/len(smoker))*100,2)) + " %")

The number of people in the study = 1338
The number of smokers = 274
The number of non-smokers = 1064
Percentage of smokers = 20.48 %
Percentage of non-smokers = 79.52 %


We can see that we have nearly 80% of population who are not smoking, so that is good sign.

Now, we will do a region based analysis and check what insights can be drawn from it.

In [29]:
unique_regions = set(region)
print(unique_regions)

{'southeast', 'southwest', 'northeast', 'northwest'}


In [47]:
#population distribution across the region
with open('insurance.csv', newline='') as insurance_csv:
    records = csv.DictReader(insurance_csv)
    southeast = []
    southwest = []
    northeast = []
    northwest = []
    for x in records:
        if x['region'] == 'southeast':
            southeast.append(x)
        elif x['region']== 'southwest':
            southwest.append(x)
        elif x['region'] == 'northeast':
            northeast.append(x)
        elif x['region']=='northwest':
            northwest.append(x)
    print("Number of people from Southeast region = " + str(len(southeast)))
    print("Number of people from Southwest region = " + str(len(southwest)))
    print("Number of people from Northeast region = " + str(len(northeast)))
    print("Number of people from Northwest region = " + str(len(northwest)))

Number of people from Southeast region = 364
Number of people from Southwest region = 325
Number of people from Northeast region = 324
Number of people from Northwest region = 325


Therefore, among the participants in the data, most come from the Southeast region. The other regions have approximately the same.

Now lets dive into trends in each region. Let's start with smokers. Which region has the highest number of smokers?

In [44]:
#Smokers perecentage across the regions
with open('insurance.csv', newline='') as insurance_csv:
    records = csv.DictReader(insurance_csv)
    smokers_se = [] #list for smokers in Southeast region
    smokers_sw = []
    smokers_ne = []
    smokers_nw = []
    for r in records:
        if r['smoker'] == 'yes':  # Check if the 'smoker' value in the record is 'yes'
            if r['region'] == 'southeast':
                smokers_se.append(r)
            elif r['region'] == 'southwest':
                smokers_sw.append(r)
            elif r['region'] == 'northeast':
                smokers_ne.append(r)
            elif r['region'] == 'northwest':
                smokers_nw.append(r)
    print("Percentage of smokers in Southeast are : " + str(len(smokers_se)*100/len(southeast)))
    print("Percentage of smokers in Southwest are : " + str(len(smokers_sw)*100/len(southwest)))
    print("Percentage of smokers in Northeast are : " + str(len(smokers_ne)*100/len(northeast)))
    print("Percentage of smokers in Northwest are : " + str(len(smokers_nw)*100/len(northwest)))


Percentage of smokers in Southeast are : 25.0
Percentage of smokers in Southwest are : 17.846153846153847
Percentage of smokers in Northeast are : 20.679012345679013
Percentage of smokers in Northwest are : 17.846153846153847


Therefore, we can see that Southeast region has the highest percentage of smokers at 25%. This could mean that average_insurance cost in the region could be higher in this region. Let's check if there is a relation.

In [52]:
#Average Insurance cost across the regions
def average_costs_region(regional):
    total_cost_regional = 0.0
    average_cost_regional = 0.0
    for rec in regional:
        total_cost_regional += float(rec['charges'])
    average_cost_regional = round(total_cost_regional/len(regional),2)
    return average_cost_regional

print("The average insurance cost for the southeast region is " + str(average_costs_region(southeast)) + " dollars.")
print("The average insurance cost for the southwest region is " + str(average_costs_region(southwest)) + " dollars.")
print("The average insurance cost for the northeast region is " + str(average_costs_region(northeast)) + " dollars.")
print("The average insurance cost for the northwest region is " + str(average_costs_region(northwest)) + " dollars.")

The average insurance cost for the southeast region is 14735.41 dollars.
The average insurance cost for the southwest region is 12346.94 dollars.
The average insurance cost for the northeast region is 13406.38 dollars.
The average insurance cost for the northwest region is 12417.58 dollars.


As expected, because of the relatively higher number of smokers in the Southeast region, the average insurance cost is higher.

Now, let's check the average age in different regions:

In [56]:
#Average Age across the regions
def average_age_region(regional):
    total_ages = 0
    average_age_regional = 0
    for person in regional:
        total_ages+= int(person['age'])
    average_cost_regional = round(total_ages/len(regional),1)
    return average_cost_regional

print("The average age of a person in the southeast region is " + str(average_age_region(southeast)) + " years.")
print("The average age of a person in the soutwest region is " + str(average_age_region(southwest)) + " years.")
print("The average age of a person in the northeast region is " + str(average_age_region(northeast)) + " years.")
print("The average age of a person in the northwest region is " + str(average_age_region(northwest)) + " years.")

The average age of a person in the southeast region is 38.9 years.
The average age of a person in the soutwest region is 39.5 years.
The average age of a person in the northeast region is 39.3 years.
The average age of a person in the northwest region is 39.2 years.


There is not much variation in average age regionally. Average age of the overall population is around 39 years, as calculated before. The regional averages are in line with this value.

Next, we can check which region has people with higher average number of children.

In [58]:
#Average number of children per person across the regions
def average_num_children(regional):
    total_num_children = 0
    average_num_childrens = 0
    for child in regional:
        total_num_children+= int(child['children'])
    average_num_childrens = round(total_num_children/len(regional),2)
    return average_num_childrens

print("The average number of children per person in the southeast region is " + str(average_num_children(southeast)))
print("The average number of children per person in the southwest region is " + str(average_num_children(southwest)))
print("The average number of children per person in the northeast region is " + str(average_num_children(northeast)))
print("The average number of children per person in the northwest region is " + str(average_num_children(northwest)))

The average number of children per person in the southeast region is 1.05
The average number of children per person in the southwest region is 1.14
The average number of children per person in the northeast region is 1.05
The average number of children per person in the northwest region is 1.15


We can see that the average number of children regionally is similar ans there isn't much variation. 

We can now check how BMI varies regionally.

In [67]:
#Average BMIs across the regions
with open('insurance.csv', newline='') as insurance_csv:
    records = csv.DictReader(insurance_csv)
    def average_bmi(regional):
        total_bmi= 0
        avg_bmi = 0
        for bmi in regional:
            total_bmi+= float(bmi['bmi'])
        avg_bmi = round(total_bmi/len(regional),2)
        return avg_bmi

    print("The average bmi in the southeast region is " + str(average_bmi(southeast)))
    print("The average bmi in the southwest region is " + str(average_bmi(southwest)))
    print("The average bmi in the northeast region is " + str(average_bmi(northeast)))
    print("The average bmi in the northwest region is " + str(average_bmi(northwest)))
   


The average bmi in the southeast region is 33.36
The average bmi in the southwest region is 30.6
The average bmi in the northeast region is 29.17
The average bmi in the northwest region is 29.2


The BMI in Southeast region is on average is higher compared to the other regions averaging at around  29-30.

Lets check the sex distribution in the different regions:




In [75]:
#Male:Female ratios across the regions
def sex_distribution(regional):
    males = 0
    females = 0
    for a in regional:
        if a['sex'] == 'male':
            males += 1
        else:
            females += 1
    return males, females

sex_distribution_se = sex_distribution(southeast)
print("The sex distribution in southeast is " + str(sex_distribution_se))
sex_distribution_sw = sex_distribution(southwest)
print("The sex distribution in southwest is " + str(sex_distribution_sw))
sex_distribution_ne = sex_distribution(northeast)
print("The sex distribution in northeast is " + str(sex_distribution_ne))
sex_distribution_nw = sex_distribution(northwest)
print("The sex distribution in northwest is " + str(sex_distribution_nw))


The sex distribution in southeast is (189, 175)
The sex distribution in southwest is (163, 162)
The sex distribution in northeast is (163, 161)
The sex distribution in northwest is (161, 164)


Sex distribtuion in Southeast shows a slightly higher number of males (189) to females (175). Other regions have an relatively similar ratios of Males to Females.

### Therefore, we can conclude that regionally speaking, the average insurance cost is higher in the southeast region. The main lever for this can be attributed to a slightly higher number of smokers as well as slightly higher bmi in the region. Another cause could also be slightly higher number of males compared to females in the region. The other attributes such as number of children and similar across all regions.

We can also analyze, the average insurance cost of a smoker vs a non-smoker. It would surely be an interesting analysis.

In [81]:
#Average Insurance cost for a smoker vs. non-smoker
with open('insurance.csv', newline='') as insurance_csv:
    records = csv.DictReader(insurance_csv)
    total_cost_smoker = 0
    n_smoker = 0
    n_nonsmoker = 0
    total_cost_non_smoker = 0
    for temp in records:
        if temp['smoker'] == 'yes':
            total_cost_smoker+=float(temp['charges'])
            n_smoker+=1
        else:
            total_cost_non_smoker+=float(temp['charges'])
            n_nonsmoker+=1
    average_cost_smoker = round(total_cost_smoker/n_smoker,2)
    average_cost_nonsmoker = round(total_cost_non_smoker/n_nonsmoker,2)
    print("Average insurance cost for a smoker is: " + str(average_cost_smoker) + " dollars.")
    print("Average insurance cost for a non-smoker is: " + str(average_cost_nonsmoker) + " dollars.")
    print("Difference in average costs between a smoker and a non smoker: " + str(average_cost_smoker - average_cost_nonsmoker) + " dollars.")
            
            

Average insurance cost for a smoker is: 32050.23 dollars.
Average insurance cost for a non-smoker is: 8434.27 dollars.
Difference in average costs between a smoker and a non smoker: 23615.96 dollars.


There is a stark difference in the average insurance costs of a smoker and a non-smoker, equivalent to 23,615.96 dollars. Thus, we can conclude that the smoking is the biggest lever when it comes to the amount of insurance cost.